holoviews/plotting/bokeh/util.py

```diff
@@ -856,6 +856,21 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
+def hold_render(f):
+    """
+    Decorator that will hold render on a Bokeh ElementPlot until after
+    the method has been called.
+    """
+    def wrapper(self, *args, **kwargs):
+        hold = self.state.hold_render
```

Comment:
This could lead to a race condition. If a trigger happens when another trigger is running. 

A()  -> starting -> `hold = False` changes `self.state.hold_render = True`
B() -> starting  -> `hold = True` changes `self.state.hold_render = True`
A() -> Finishes -> changes `self.state.hold_render = False`
A() -> Finishes -> changes `self.state.hold_render = True`

Could we use an integer and add and subtract here?

Reply:
> will therefore always restore `hold_render=False`

It will restore to what `hold` was when the function started. In my example, `B` would have `hold = True` because `A` had set it so. 

> It's not like there's anything async happening

We are getting closer and closer to a no-gil Python world. 

Maybe I'm just too cautious here or not understanding the logic well enough :upside_down_face:

Reply:
Yeah, I don't think there's an issue here. Certainly none of our code would be ready to perform plot updates on different threads.